File: icarus.cpp
// Problem: https://csacademy.com/ieeextreme-practice/task/icarus
// Solution by: Osama Khallouf (Exeedo)

#include <bits/stdc++.h>
using namespace std;

string moves;

void solve() {
    int count_r = 0, count_u = 0;
    for (char move : moves) {
        if(move == 'R') {
            count_r++;
        }
        if (move == 'U') {
            count_u++;
        }
    }
    int n = max(count_u, count_r) + 1;
    if (count_u == 0 || count_r == 0) {
        n = 2;
    }
    int start = 1;
    int end = n;
    if (count_r >= count_u) {
        start = n;
        end = 1;
    }
    cout << n << ' ' << start << ' ' << end << endl;
    for (int i = 1; i < n; i++) {
        cout << "0 " << i + 1 << endl;
    }
    cout << "0 0\n";
}

int main()
{
    cin >> moves;
    solve();
	return 0;
}

File: icarus.py
# Problem: https://csacademy.com/ieeextreme-practice/task/icarus
# Solution by: Osama Khallouf (Exeedo)

s = input()
count_u = s.count('U')
count_r = s.count('R')

n = max(count_r, count_u) + 1
if count_r == 0 or count_u == 0:
    n = 2

start = 1
end = n
if count_r >= count_u:
    start = n
    end = 1

print(f"{n} {start} {end}")
for node in range(1, n):
    print(f"0 {node+1}")
print("0 0")